source/beamline/IDEAS/IDEAS13ElementGeDetector.h

```diff
@@ -51,8 +51,39 @@ class IDEAS13ElementGeDetector : public AMXRFDetector
 	/// The vortex detectors support elapsed time.
 	virtual bool supportsElapsedTime() const { return true; }
 
+	/// Returns the real time for the detector.
+	AMDetector *dwellTime() const {return ge13ElementRealTime_; }
+
+    /// Returns AMPVControl for peaking time
+    double peakingTime() const { return peakingTimeControl_->value(); }
+    /// Returns AMPVControl for preamp gain
+    double preampGain1() const  { return preampGainControl1_->value(); }
```

Comment:
I would change this to `double preampGainAt(int index) const` and use the list method I described.